web_external/js/views/body/LayersPanel.js

```diff
@@ -18,20 +19,87 @@ minerva.views.LayersPanel = minerva.View.extend({
+        var datasetId = $(event.currentTarget).attr('m-dataset-id');
+        var option = $(event.currentTarget).attr('m-order-option');
+        var dataset = this.collection.get(datasetId);
+
+        var currentDatasetIndex = _.indexOf(this.collection.models, dataset);
+        var displayedDatasets = _.filter(this.collection.models, function (set) {
+            return set.get('displayed');
+        });
+
+        var numberOfPossibleLayers = this.collection.models.length;
+        var numberOfLayersInSession = displayedDatasets.length;
+
+        if (displayedDatasets[currentDatasetIndex - 1]) {
+            prevDataset = displayedDatasets[currentDatasetIndex - 1];
+        }
+
+        if (displayedDatasets[currentDatasetIndex + 1]) {
+            nextDataset = displayedDatasets[currentDatasetIndex + 1];
+        }
+
+        var stackValues = _.map(this.collection.models, function (dataset) {
+            return dataset.get('stack');
+        });
+
+        var currentStack = dataset.get('stack');
+        // Retrieve the first and last stack value in the collection
+        var lastValueInStack = _.last((stackValues).sort(function(a, b) {
+            return a - b;
+        }));
+        var firstValueInStack = _.first((stackValues).sort(function(a, b) {
+            return a - b;
+        }));
+
+        if (option === 'moveToTop' && currentStack !== lastValueInStack) {
+            dataset.set('stack', lastValueInStack + 1);
+        } else if (option === 'moveToBottom' && currentStack !== (numberOfPossibleLayers - numberOfLayersInSession)) {
+            dataset.set('stack', firstValueInStack - 1);
+        } else if (option === 'moveUp' && currentStack !== lastValueInStack) {
+            dataset.set('stack', currentStack + 1);
+            (nextDataset || prevDataset).set('stack', currentStack);
+        } else if (option === 'moveDown' && currentStack !== (numberOfPossibleLayers - numberOfLayersInSession)) {
+            dataset.set('stack', currentStack - 1);
+            (prevDataset || nextDataset).set('stack', currentStack);
+        }
+
+        // Re-set the layer order on the map and then set the new order
+        dataset.set('order', null);
+        dataset.set('order', option);
+    },
+
     initialize: function (settings) {
         settings = settings || {};
         this.collection = settings.collection;
-        this.listenTo(this.collection, 'change:displayed', function () {
-            this.render();
+        this.layersOrderOptions = [
+            {'title': 'move up', 'method': 'moveUp', 'class': 'up'},
+            {'title': 'move down', 'method': 'moveDown', 'class': 'down'},
+            {'title': 'move to top', 'method': 'moveToTop', 'class': 'double-up'},
+            {'title': 'move to bottom', 'method': 'moveToBottom', 'class': 'double-down'}
+        ];
+
+        this.listenTo(this.collection, 'change:displayed change:order', function (dataset) {
+            this.render(dataset);
         }, this);
```

Comment:
since dataset is getting removed from the render call below, you should remove it from the call here and from the function definition.